cpp/src/layout/legacy/barnes_hut.cuh

```diff
@@ -162,9 +165,19 @@ void barnes_hut(raft::handle_t const& handle,
   sort(graph, stream_view.value());
   RAFT_CHECK_CUDA(stream_view.value());
 
-  graph.degree(massl, cugraph::legacy::DegreeDirection::OUT);
+  graph.degree(massl_et, cugraph::legacy::DegreeDirection::OUT);
   RAFT_CHECK_CUDA(stream_view.value());
 
+  // Cast massl from edge_t to int.
```

Comment:
We should probably look at updating the functions to use `edge_t`.  While it is uncommon, it is possible for an edge in a 64-bit graph to have > 2^31-1 neighbors, in which case we'll generate an incorrect answer by down casting this into a 32-bit integer.  Not sure the ripple effects here... my first glance is that it seems tractable.

Reply:
Thanks, working on this now! It's straightforward so far